### ISSUE TITLE
Too many background pixels of tanks and temples

### ISSUE THREAD

**Issue by flamehaze1115** [alibaba/cascade-stereo#14]
Hi. Thanks very much for your great work and releasing code. I use your pretrained model to test on tanks and temple dataset, the final fused model has too many background pixels such as sky in LIghtHouse. But in your paper, the model is quite clean. So how to remove such background pixels?

**Comment by Txusheng** [comment]
> Hi. Thanks very much for your great work and releasing code. I use your pretrained model to test on tanks and temple dataset, the final fused model has too many background pixels such as sky in LIghtHouse. But in your paper, the model is quite clean. So how to remove such background pixels?

Hi. Do you also fuse the depth maps using the fusibile method provided by Gipuma? It seems that for datasets provided in Tanks and Temples, it takes more than 11G memory of GPU to fuse the depth maps with resolution of 1920x1080.

**Comment by tatsy** [comment]
Hi. I encountered the same problem of the large memory consumption mentioned by @Txusheng for Tanks and Temples dataset. Particularly, I failed to fuse the depth maps for some image sets such as Francis and Lighthouse even by using a GPU with 48GB memory. Does anyone know a workaround for that?

**Comment by tatsy** [comment]
For reference, I consumed about 22GB memory by Fusibile for processing Francis, Lighthouse and some other image sets in Tanks and Temples.

**Comment by TruongKhang** [comment]
@tatsy I used fusibile to produce the 3D models on Tanks and Temples. However, for some scenes such as Lighthouse, Francis, Playground, there are too many noise pixels in the background. Did you have the same problem? Can you tell me your settings for CasMVSNet and fusibile to produce a good result?

Thanks,
Khang

**Comment by tatsy** [comment]
Hi @TruongKhang,

I tested both the fusibile and python fusion. As far as I tested, the amount of noise is a bit less by python fusion than by fusibile. I also found that we can reduce the noise by increasing the number of views for the MVSNet part and also by increasing the number of consistent views for the depth fusion.

However, I have not yet completely solved the problem of much noise in the background, and have not yet succeeded in reproducing the results for Tanks & Temples reported in their paper.

I'd really appreciate for it if you would share us your knowledge to solve the problem.

**Comment by TruongKhang** [comment]
Hi @tatsy ,
I've been testing with both the provided python fusion and the fusibile fusion. As you said, the python fusion is much better than fusibile fusion for the Tanks & Temples dataset. Also, we need to run with a short depth range which is already provided in the dataset. I think the result is acceptable now.

**Comment by tatsy** [comment]
Hi @TruongKhang,

Thank you for sharing your knowledge. Yes, I got almost the same results as you got. I also used the short depth range provided by the MVSNet repo. Maybe, the mean score for the intermediate image set will be around 50 or slightly more than that. Is it also true for you?

**Comment by TruongKhang** [comment]
@tatsy , did you try to submit those results to the Tank & Temples website? And you got the lower accuracy as reported in the paper? I haven't tried to submit yet.

**Comment by YuhsiHu** [comment]
I get terrible point cloud results on tank and temples dataset using pre-trained model. Almost nothing can be reconstruct. I used 
--dataset=general_eval --batch_size=1 --testpath=$TESTPATH  --testlist=$TESTLIST --loadckpt $CKPT_FILE --outdir $save_results_dir  --interval_scale 1.06 --max_h=2048 --max_w=2048${@:2}
What other params should I set?

**Comment by TruongKhang** [comment]
@YuhsiHu , which fusion method did you use? I suggest you should change the number of consistent views.

**Comment by ChenLiufeng** [comment]
> Hi @TruongKhang,
> 
> Thank you for sharing your knowledge. Yes, I got almost the same results as those you got. I used the short depth range provided by the MVSNet repo, too. Maybe, the mean score for the intermediate image set will be around 50 or slightly more than that. Is it also true for you?
Hi@tatsy  how to set  the short depth range?

**Comment by ChenLiufeng** [comment]
> @YuhsiHu , which fusion method did you use? I suggest you should change the number of consistent views.

@TruongKhang  the number of consistent views , how many is better?

**Comment by TruongKhang** [comment]
@ChenLiufeng  It is changed manually for each scene from my testing. It is higher than 5 views. And I think in their submission, they use both the fusibile fusion and the basic fusion provided in their source code.

**Comment by ChenLiufeng** [comment]
> @ChenLiufeng It is changed manually for each scene from my testing. It is higher than 5 views. And I think in their submission, they use both the fusibile fusion and the basic fusion provided in their sou

> @ChenLiufeng It is changed manually for each scene from my testing. It is higher than 5 views. And I think in their submission, they use both the fusibile fusion and the basic fusion provided in their source code.
@TruongKhang  Thank you for your applying. I change ' --max_h=2048 --max_w=2048' ,test on family.   I can't see the statue in .ply.  Can you show me the details? What can I do to improve my results？

**Comment by TruongKhang** [comment]
@ChenLiufeng did you use the camera parameters with a short-range depth? Can you show me a picture of your result?
There are some parameters you should notice:
1. **--filter_method**: you can set to "gipuma" to use fusibile fusion. If you want to use the basic fusion then just ignore this parameter.
2. If you use the basic fusion, you can try to change the parameter **--thres_view**. Here I recommend setting it to 5. You can test with the other values.
3. If you use the fusibile fusion, you should notice two parameters **--disp_threshold** and **num_consistent**. When you increase disp_threshold, you should increase num_consistent. Here you can test with disp_threshold = 0.4 and num_consistent.
I also run several other MVS methods, the confidence parameter **prob_threshold** or **conf** sometimes has a good impact to remove outliers in the background. But in this CasMVSNet, it doesn't affect much!
Hope it help. Let me know if you have any problems!

**Comment by wangchengze001** [comment]
hello,how to chang the code test.py , in order to eval T&T datasets ? thank you

**Comment by xiaomingHUST** [comment]
> 

@tatsy sorry to bother you. i use the preprocessed data that MVSnet offered,is it different from yours? what's the short depth range data that you mentioned.
thanks a lot.

**Comment by zhao-you-fei** [comment]
请问我按照casmvsnet流程测试T&T数据集出现以下效果该怎么有效解决,filter-method默认normal
最终融合:
![image](https://user-images.githubusercontent.com/75787742/196885115-1b957f2f-7617-4d1d-89d1-3be183a48ebf.png)
ply_local文件夹点云:
![image](https://user-images.githubusercontent.com/75787742/196885175-74ddda6a-fdf0-4510-bb6b-08bd8653f1b0.png)
深度图:
![image](https://user-images.githubusercontent.com/75787742/196885253-267568bd-f785-4f00-93db-4cefa71802c8.png)

**Comment by wangchengze001** [comment]
这明显相机内外参不对，主要调整下可微单应性变化那的外参矩阵



---Original---
From: ***@***.***&gt;
Date: Thu, Oct 20, 2022 15:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alibaba/cascade-stereo] Too many background pixels of tanks andtemples (#14)




 
请问我按照casmvsnet流程测试T&amp;T数据集出现以下效果该怎么有效解决,filter-method默认normal
 最终融合:
 
 ply_local文件夹点云:
 
 深度图:

**Comment by WangDongBUAA** [comment]
> 请问我按照casmvsnet流程测试T&T数据集出现以下效果该怎么有效解决,filter-method默认normal 最终融合: ![image](https://user-images.githubusercontent.com/75787742/196885115-1b957f2f-7617-4d1d-89d1-3be183a48ebf.png) ply_local文件夹点云: ![image](https://user-images.githubusercontent.com/75787742/196885175-74ddda6a-fdf0-4510-bb6b-08bd8653f1b0.png) 深度图: ![image](https://user-images.githubusercontent.com/75787742/196885253-267568bd-f785-4f00-93db-4cefa71802c8.png)

你好，我也遇到了这样的问题，请问你解决了吗
